duckduckgo/BrowserServicesKit
# Use Hash instead of etag as allowlist and temp list identifiers

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: 
 - https://app.asana.com/0/1163321984198618/1204236089135865/f
 - https://app.asana.com/0/414709148257752/1204916184448572/f
 
iOS PR: 
macOS PR: 
What kind of version bump will this require?: Patch

**Optional**:

Tech Design URL:
CC:

**Description**:

Use Hash instead of etag as allowlist and temp list identifiers to avoid excess rule recompilation.



**Steps to test this PR**:

Generic smoke tests of content blocking feature.

Also:
1. Disable using fetched config by:
    - Passing nil to PrivacyConfigurationManager first two parameters.
    - Comment out call to PrivacyConfigurationManager.reload(...)
2. Put breakpoints in `ContentBlockingRulesCompilationTask` on `lookUpContentRuleList` and `compileContentRuleList`.
3. Run app few times - you should see restarts does not recompile anything.
4. Change embedded config hash for allowlist, or, change unprotected domains contents, or, change content blocking exceptions (not hash) and see that it triggers recompilation.
5. Change embedded etag in `AppPrivacyConfigurationDataProvider` - this should not cause recompilation.



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
